### PR TITLE
Stop using finalizers if not extrictly needed

### LINKED ISSUE #314: e2e: testing operator uninstall
We should test uninstallation of `elemental-operator`, esp. the removal of associated CRDs.

### PR DESCRIPTION
**Pull request by davidcassany**
Fixes #314

### CODE REVIEW

**Comment by davidcassany** [comment]
This PR _fixes_ #314 in the sense that elemental-operator can be uninstalled and reinstalled without causing weird effects neither leaving any resource managed by the elemental-operator behind.

However under some circumstances this is still likely to lead to states which are hard to recover from without redeploying nodes and similar procedures. As an example, if the elemental-operator is uninstalled when having a deployed elemental cluster this causes the cluster to enter in updating state and causes the machine pool to enter in failed stated (as it is linked to a machineinventory and machineinventoryselector that are no longer available). I doubt this can be cleanly solved without redeploying the cluster. As it is not possible to regenerate equivalent machineinventory for the already existing node, as this is tied to a registry end point that was also destroyed.

In short, `helm uninstall` works and a follow up `helm install` works too, however for a proper uninstall the admin should first delete any elemental cluster and/or any other related rancher resource.
